src/main/java/numerator/Parser.java

```diff
@@ -0,0 +1,202 @@
+package numerator;
+
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import numerator.exceptions.NumeratorException;
+import numerator.exceptions.parser.InputFormatException;
+import numerator.exceptions.parser.InputNotRecognisedException;
+import numerator.exceptions.parser.TaskIndexOutOfBoundsException;
+import numerator.task.Task;
+import numerator.task.TaskList;
+
+
+/**
+ * Parses the input from the user and performs the corresponding action
+ */
+public class Parser {
+
+    public static final String BYE_STRING = "Bye! Just close the window to exit";
+
+    private static final String INPUT_BYE = "bye";
+    private static final String INPUT_MARK = "mark";
+    private static final String INPUT_UNMARK = "unmark";
+    private static final String INPUT_DELETE = "delete";
+    private static final String INPUT_TODO = "todo";
+    private static final String INPUT_DEADLINE = "deadline";
+    private static final String INPUT_EVENT = "event";
+    private static final String INPUT_LIST = "list";
+    private static final String INPUT_FIND = "find";
+
+
+    /**
+     * Parses the input and performs the corresponding action
+     *
+     * @param input    the input from the user
+     * @param taskList the list of tasks
+     * @param storage  the storage object
+     * @return the response to the user
+     * @throws NumeratorException if the input is not recognised
+     */
+    public static String parseArguments(String input, TaskList taskList, Storage storage) throws NumeratorException {
```

Comment:
variable name for TaskList could be plural therefore you might consider using tasks instead of taskLIst